engine/classes/ElggFile.php

```diff
@@ -154,9 +154,10 @@ public function detectMimeType($file = null, $default = null) {
 			$mime = mime_content_type($file);
 		}
 
+		$original_filename = isset($this) ? $this->originalfilename : basename($file);
```

Comment:
Can you add a `@note` in the phpdoc that we're calling this both dynamically and statically?